CDTS Blazor/Data/Services/AzureKeyVaultService.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace CDNApplication.Data.Services
+{
+    using System.Threading.Tasks;
+    using Microsoft.Azure.KeyVault;
+    using Microsoft.Azure.KeyVault.Models;
+    using Microsoft.Azure.Services.AppAuthentication;
+    using Microsoft.Rest.Azure;
+
+    /// <summary>
+    ///  Represents the Azure key vault service.
+    /// </summary>
+    public class AzureKeyVaultService
+    {
+        private string dNs;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureKeyVaultService"/> class.
+        /// </summary>
+        /// <param name="dNS">DNS for the key vault.</param>
+        public AzureKeyVaultService(string dNS)
+        {
+            this.dNs = dNS;
+        }
+
+        /// <summary>
+        /// Gets a secret from azure key vault.
+        /// </summary>
+        /// <param name="secretName">Secret to get.</param>
+        /// <returns>Secret value.</returns>
+        public string GetSecret(string secretName)
+        {
+            using (var keyVaultClient = GetKeyVaultClient())
+            {
+                var sercret = keyVaultClient.GetSecretAsync(vaultBaseUrl: this.dNs, secretName: secretName).ConfigureAwait(true).GetAwaiter().GetResult();
+
+                return sercret.Value;
+            }
+        }
+
+        /// <summary>
+        /// Get a list of secrets from the azure key vault.
+        /// </summary>
+        /// <returns>List of secrets.</returns>
+        public IPage<SecretItem> GetListOfSecretsAsync()
```

Comment:
Considering renaming or adjusting the function to return a list of secrets. You can remove ther Async at the end of the function name since the caller doesn't need to be aware of the details of how the function works. Right now the function is misleading since it doesn't return a List<SecretItem> but a IPage<SecretItem>.
```suggestion
        public IPage<SecretItem> GetListOfSecretsAsync()
```